pytential/symbolic/compiler.py

```diff
@@ -465,44 +464,46 @@ def execute(self, exec_mapper, pre_assign_check=None):
 # {{{ compiler
 
 class OperatorCompiler(IdentityMapper):
-    def __init__(self, places, prefix="_expr",
-            max_vectors_in_batch_expr=None):
-        IdentityMapper.__init__(self)
+    def __init__(self,
+            places,
+            prefix: str = "_expr",
+            max_vectors_in_batch_expr: Optional[int] = None) -> None:
+        super().__init__()
+
         self.places = places
         self.prefix = prefix
-
         self.max_vectors_in_batch_expr = max_vectors_in_batch_expr
 
-        self.code = []
+        self.code: List[Instruction] = []
         self.expr_to_var = {}
+        self.assigned_names: Set[str] = set()
+        self.group_to_operators: Dict[Hashable, Set[IntG]] = {}
 
-        self.assigned_names = set()
-
-    def op_group_features(self, expr):
+    def op_group_features(self, expr) -> Hashable:
         from pytential.symbolic.primitives import hashable_kernel_args
         lpot_source = self.places.get_geometry(expr.source.geometry)
         return (
                 lpot_source.op_group_features(expr)
                 + hashable_kernel_args(expr.kernel_arguments))
 
     @memoize_method
-    def dep_mapper_factory(self, include_subscripts=False):
-        self.dep_mapper = DependencyMapper(
-                #include_operator_bindings=False,
+    def dep_mapper_factory(self, include_subscripts: bool = False):
+        return DependencyMapper(
+                # include_operator_bindings=False,
                 include_lookups=False,
                 include_subscripts=include_subscripts,
                 include_calls="descend_args")
 
-        return self.dep_mapper
+    @property
+    def dep_mapper(self):
+        return self.dep_mapper_factory()
```

Comment:
This is probably a behavior change, but I didn't really understand what the intention of the original code was.

It gets memoized, but it also sets that `dep_mapper` attribute based on `include_subcripts` (?)

Reply:
I agree that the old code looks might fishy.